src/Mod/TechDraw/App/DrawViewPart.cpp

```diff
@@ -372,7 +371,7 @@ TechDraw::GeometryObject* DrawViewPart::buildGeometryObject(TopoDS_Shape& shape,
         //https://github.com/KDE/clazy/blob/1.11/docs/checks/README-connect-3arg-lambda.md
         connectHlrWatcher = QObject::connect(&m_hlrWatcher, &QFutureWatcherBase::finished,
                                              &m_hlrWatcher, [this] { this->onHlrFinished(); } );
-        m_hlrFuture = QtConcurrent::run(go, &GeometryObject::projectShape, shape, viewAxis);
+        m_hlrFuture = QtConcurrent::run(*go, &GeometryObject::projectShape, shape, viewAxis);
```

Comment:
This line (374) causes the geometry object to not be updated.  If I change to        m_hlrFuture = QtConcurrent::run(go.get(), &GeometryObject::projectShape, shape, viewAxis);
Then everything works again.   Does using the raw pointer here defeat what you are doing with the shared pointer?

Reply:
The `operator*` actually returns a reference while the get() method returns a pointer. Is there a unit test in TD that covers this function?

Reply:
I have force-pushed the fixed branch.

Reply:
On [SO](https://stackoverflow.com/questions/37298600/qtconcurrent-wait-for-finished-when-app-is-about-to-quit) this method has been suggested:
`connect(qApp, &QCoreApplication::aboutToQuit, qApp, []() { QThreadPool::globalInstance()->waitForDone(); });`
But calling `QThreadPool::globalInstance()->waitForDone()` while the QtConcurrent::run is still running leads to a crash.

But what seems to work fine is running a second event loop for a short while that blocks the main thread and gives the worker thread enough time to finish its work:
```
from PySide import QtCore


doc = FreeCAD.newDocument("TDHatch")

# make source feature
box = doc.addObject("Part::Box", "Box")

page = doc.addObject('TechDraw::DrawPage', 'Page')

template = doc.addObject('TechDraw::DrawSVGTemplate', 'Template')
template.Template = FreeCAD.getResourceDir() + 'Mod/TechDraw/Templates/A4_LandscapeTD.svg'

page.Template = template
doc.recompute()
view = doc.addObject('TechDraw::DrawViewPart', 'View')
view.Source = [box]

page.addView(view)
view.X = 100
view.Y = 100

# returns empty list
view.getVisibleEdges()
doc.recompute()

# still returns empty list
view.getVisibleEdges()


loop = QtCore.QEventLoop()

timer = QtCore.QTimer()
timer.setSingleShot(True)
timer.timeout.connect(loop.quit)

timer.start(500)
loop.exec_()

# finally returns a non-empty list
view.getVisibleEdges()
```